clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/view/GoogleLoadBalancerProvider.groovy

```diff
@@ -53,21 +53,32 @@ class GoogleLoadBalancerProvider implements LoadBalancerProvider<GoogleLoadBalan
 
   @Override
   Set<GoogleLoadBalancerView> getApplicationLoadBalancers(String application) {
-    def pattern = Keys.getLoadBalancerKey("*", "*", "${application}*")
-    def identifiers = cacheView.filterIdentifiers(LOAD_BALANCERS.ns, pattern)
-
-    def applicationServerGroups = cacheView.getAll(
+    String pattern = Keys.getLoadBalancerKey("*", "*", "${application}*")
+    Collection<String> identifiers = cacheView.filterIdentifiers(LOAD_BALANCERS.ns, pattern)
+
+    // It is possible to configure a server group in application A to use a load balancer from
+    // application B. Therefore, if (and only if) we have not already retrieved load balancer
+    // identifiers for all applications, we need to retrieve identifiers for every load balancer
+    // associated with a server group from this application.
+    if (application != null && !application.isEmpty()) {
```

Comment:
I'd usually static import Guava's Strings#isNullOrEmpty for this, so it's just
  if (!isNullOrEmpty(application))

Reply:
What a great util, thanks Plumpy and guava!
![national-gardens-10-giant-thailand-guava-fruit-original-imaej9hxrudux8sp](https://user-images.githubusercontent.com/15936279/61223303-5cbda000-a6ea-11e9-9cce-7b8c9fa3a668.jpeg)